cmd/juju/application/deploy.go

```diff
@@ -185,12 +188,15 @@ func newDeployCommand() *DeployCommand {
 		return modelconfig.NewClient(api)
 	}
 	deployCmd.NewDownloadClient = func() (store.DownloadBundleClient, error) {
-		apiRoot, err := deployCmd.ModelCommandBase.NewAPIRoot()
-		if err != nil {
-			return nil, errors.Trace(err)
+		if deployCmd.apiRoot == nil {
```

Comment:
Probably we could wrap these simple cache logic into a helper method `deployCmd.getAPIRoot()`
and